src/ui/map.js

```diff
@@ -2734,19 +2746,30 @@ class Map extends Camera {
         // Even though `_styleDirty` and `_sourcesDirty` are reset in this
         // method, synchronous events fired during Style#update or
         // Style#_updateSources could have caused them to be set again.
-        const somethingDirty = this._sourcesDirty || this._styleDirty || this._placementDirty;
+        const somethingDirty = this._sourcesDirty || this._styleDirty || this._placementDirty || averageElevationChanged;
         if (somethingDirty || this._repaint) {
             this.triggerRepaint();
         } else {
-            this._triggerFrame(false);
-            if (!this.isMoving() && this.loaded()) {
-                this.fire(new Event('idle'));
-                this._isInitialLoad = false;
-                // check the options to see if need to calculate the speed index
-                if (this.speedIndexTiming) {
-                    const speedIndexNumber = this._calculateSpeedIndex();
-                    this.fire(new Event('speedindexcompleted', {speedIndex: speedIndexNumber}));
-                    this.speedIndexTiming = false;
+            const willIdle = !this.isMoving() && this.loaded();
```

Comment:
Is this check still valid in infinite paint mode? e.g. when we have map.repaint = true

Reply:
I checked, and this branch is not reached when `repaint = true`. When `repaint = true`, it samples continuously at the 500ms timeout interval.